src/tactic/lint/type_classes.lean

```diff
@@ -116,24 +117,26 @@ if ty = `(Prop) then pure none else do
 `(Sort _) ← whnf ty | pure none,
 insts ← attribute.get_instances `instance,
 insts_tys ← insts.mmap $ λ i, expr.pi_codomain <$> declaration.type <$> get_decl i,
-let inhabited_insts := insts_tys.filter (λ i,
-  i.app_fn.const_name = ``inhabited ∨ i.app_fn.const_name = `unique),
-let inhabited_tys := inhabited_insts.map (λ i, i.app_arg.get_app_fn.const_name),
-if d.to_name ∈ inhabited_tys then
+let nonempty_insts := insts_tys.filter (λ i,
+  i.app_fn.const_name = ``nonempty
+  ∨ i.app_fn.const_name = ``inhabited
+  ∨ i.app_fn.const_name = `unique),
```

Comment:
```suggestion
  i.app_fn.const_name ∈ [``nonempty, ``inhabited, `unique]),
```
(untested)

Reply:
I'd thought of doing that, but for lint speed reasons I thought it might be better to leave it expanded out. Maybe my code is a micro-optimization that's not worth doing -- do you think I should switch it to the more-easily-extended list?

Reply:
Somewhat counterintuitively, looks like `∈` is actually faster in this very unscientific toy example!

```lean
import tactic

set_option profiler true

run_cmd do -- 1.81s
  e ← tactic.get_env,
  tactic.trace $ e.fold (0 : ℕ) $ λ d n,
    let i := d.type in
    if i.app_fn.const_name = ``nonempty
      ∨ i.app_fn.const_name = ``inhabited
      ∨ i.app_fn.const_name = `unique
    then n + 1
    else n

run_cmd do -- 1.41s
  e ← tactic.get_env,
  tactic.trace $ e.fold (0 : ℕ) $ λ d n,
    let i := d.type in
    if i.app_fn.const_name ∈ [``nonempty, ``inhabited, `unique]
    then n + 1
    else n
```

Reply:
Thanks for the unscientific test! I switched it over.